include/GafferScene/SceneAlgo.h

```diff
@@ -73,6 +74,19 @@ void matchingPaths( const Filter *filter, const ScenePlug *scene, PathMatcher &p
 /// FilteredSceneProcessor::filterPlug() would be passed.
 void matchingPaths( const Gaffer::IntPlug *filterPlug, const ScenePlug *scene, PathMatcher &paths );
 
+/// Calls a functor on all paths in the scene
+/// The functor must take ( const ScenePlug*, const ScenePlug::ScenePath& ), and can return false to prune traversal
+template <class ThreadableFunctor>
+void parallelTraverse( const ScenePlug *scene, ThreadableFunctor &f, const Gaffer::Context *context = Gaffer::Context::current() );
+
+/// Calls a functor on all paths in the scene that are matched by the filter.  The functor must take ( const ScenePlug*, const ScenePlug::ScenePath& ), and its return value is ignored
+template <class ThreadableFunctor>
+void filteredParallelTraverse( const Filter *filter, const ScenePlug *scene, ThreadableFunctor &f, const Gaffer::Context *context = Gaffer::Context::current() );
```

Comment:
How about we put the filter argument after the scene, so it reads more naturally as "like a `parallelTraverse()`, but with an extra bit"? Also, can we not keep the same semantics for the return value of the functor? So traversal is pruned if either the functor requests it or if the filter has no descendant matches?